org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/SemanticHighlightReconcilerStrategy.java

```diff
@@ -254,10 +256,12 @@ private void cancelSemanticTokensFull() {
 	}
 
 	private void fullReconcile() {
-		if (disabled || !isInstalled) { // Skip any processing
+		IDocument theDocument = document;
+		var ts = DocumentUtil.getDocumentModificationStamp(theDocument);
+		if (disabled || !isInstalled || ts == timestamp) { // Skip any processing
 			return;
 		}
-		IDocument theDocument = document;
+		timestamp = ts;
```

Comment:
Now that I understand your PR better, should we not have the check these two methods?

```
	@Override
	public void reconcile(final DirtyRegion dirtyRegion, final IRegion subRegion) {
		fullReconcile();
	}

	@Override
	public void reconcile(final IRegion partition) {
		fullReconcile();
	}
```

In the end it will end the reconciliation no matter where we add the check and the exit, but I think it makes more clear what it the purpose of the check and exit

Reply:
Okay, do mean something like this?:
```java
	@Override
	public void reconcile(final DirtyRegion dirtyRegion, final IRegion subRegion) {
                var ts = DocumentUtil.getDocumentModificationStamp(document);
               if (ts != timestamp) {
		  fullReconcile();
                  timestamp = ts;
                }
	}
```